pkg/apis/batch/v1/defaults.go

```diff
@@ -62,6 +62,15 @@ func SetDefaults_Job(obj *batchv1.Job) {
 			}
 		}
 	}
+	if obj.Spec.PodReplacementPolicy == nil {
```

Comment:
@mimowo adding these defaulting fields causes roundtrip tests to fail.  Not sure if these means that we can't actually add default values for string enums after all.  

I ran into these issues before when trying to add defaults and I'm not sure I can default `PodReplacementPolicy` types.

Reply:
We answered this offline.  Defaults was removed and defaulting logic will be interpreted from nil.